ome/omero-cli-render
# pip installable

In this PR:

* Make the render plugin installable via pip
* Add structure to run the integration tests in travis
* flake8 is not unable yet (due to ``cli.py``)
* Make the plugin usable with 5.4.0

Few commits have not been ported to [openmicroscopy/develop](https://github.com/openmicroscopy/openmicroscopy) so some methods cannot be used yet.

It is not currently possible to run the CLI tests outside [openmicroscopy](https://github.com/openmicroscopy/openmicroscopy)(https://github.com/openmicroscopy/openmicroscopy) repo without copying ``cli.py`` to this repo.
When ``cli.py`` is available from ``testlib``, the file should be removed from this repository and the test adjusted.

2 PRs will be opened against  [openmicroscopy/develop](https://github.com/openmicroscopy/openmicroscopy):
* One PR to port the commits from metadata branch. See https://github.com/openmicroscopy/openmicroscopy/pull/5545
* One PR migrating ``cli.py`` https://github.com/openmicroscopy/openmicroscopy/pull/5548

To Test:

manually

 * Install on an existing omero installation by running ``python setup.py sdist install``
 *  Check the commands of the plugin i.e. ``bin/omero render -h`` for the list of supported commands

Automatically
 * Check the output of the travis build. Check that the plugin is installed and the tests are green

Similar strategy could be used on other plugin repositories to run the integration tests when we have migrated ``cli.py``

cc @joshmoore @manics
